docs/specs/sdk.md

```diff
@@ -23,6 +23,116 @@ It is more preferable to have ephemeral refresh token that are only stored in me
+// Otherwise, it behaves like authorize().
+function reauthenticate(options: AuthorizeOptions): Promise<{ userInfo: UserInfo, state?: string }>;
+
+// canReauthenticate returns true if the user has logged in and they have authenticator to reauthenticate themselves.
+// Note that there are users who CANNOT reauthenticate themselves.
+// For example, a user who signed up with Google does not have any authenticator thus cannot reauthenticate themselves.
+// Before you want to trigger reauthentication, you must first call this function to see if reauthentication is possible.
+// Otherwise, your user see will an error message if you trigger reauthentication.
+function canReauthenticate(): boolean;
+
+// Always authenticate the end-user fully.
+// This is the suitable user experience when the user taps "Sign in" or "Sign up"
+const options: AuthorizeOptions = {
+  redirectURI: "myapp://host/path",
+  prompt: "login",
+};
+
+// Authenticate the end-user, allow reusing existing session.
+const options: AuthorizeOptions = {
+  redirectURI: "myapp://host/path",
+};
+
+// Always reauthenticate the current user.
+// You MUST first decode the ID token and check if the claim `https://authgear.com/user/can_reauthenticate` is true,
+// otherwise your user WILL see an error.
+const options: AuthorizeOptions = {
+  redirectURI: "myapp://host/path",
+  prompt: "login",
+  maxAge: 0,
+};
+
+// Reauthenticate the current user if last authentication is not within 5 minutes.
+// You MUST first decode the ID token and check if the claim `https://authgear.com/user/can_reauthenticate` is true,
+// otherwise your user WILL see an error.
+const options: AuthorizeOptions = {
+  redirectURI: "myapp://host/path",
+  prompt: "login",
+  maxAge: 60 * 5,
+};
+
+// After reauthentication, get the ID token from the SDK and send it along with your sensitive request.
+// Your server is responsible for verifying the signature of the ID token and validating the `auth_time`.
+authgear.fetch("https://api.myserver.com/v1/edit-profile", {
+  method: "POST",
+  headers: {
+    "content-type": "application/json",
+  },
+  body: JSON.stringify({
+    profile: { ... },
+    id_token: authgear.getIDTokenHint(),
+  }}),
+});
+
+// Example of how to trigger reauthentication before performing sensitive operation.
+async function onClickSave() {
+  const canReauthenticate = authgear.canReauthenticate();
+
+  // The user cannot be reauthenticated.
+  // However, we do not want to block them from performing sensitive operation.
+  // So we still allow them to proceed.
+  if (!canReauthenticate) {
+    // You must send the ID token to your server.
+    // Your server must verify the signature and check the claims inside.
+    // Your server should allow user to skip reauthentication if it is impossible.
+    await performSensitiveOperation(authgear.getIDTokenHint());
+    return;
+  }
+
+  const options: AuthorizeOptions = {
+    redirectURI: "myapp://host/path",
+    prompt: "login",
```

Comment:
I don't think prompt and maxAge is needed for reauthenticate?

Reply:
maxAge may be needed sometimes. When the app has a series of sensitive operation to perform, it should be acceptable that the user has reauthenticated themselves recently enough, say 5 minutes. So in this case, the developer should specify a maxAge of `5 * 60`, to provide a better UX. The user does not need to go through reauthenticate themselves repeatedly.

Reply:
@louischan-oursky thanks got it.

What about prompt? There are no way prompt is changable in reauthenticate?

Reply:
Yes, you're correct. I have removed `prompt` from ReauthenticateOptions.